internal/commands/gen.py

```diff
@@ -216,6 +217,8 @@ def command_gen(
             with open(context.path.testcases_hashes, "r") as f:
                 official_testcase_hashes: dict[str, str] = json.load(f)
             formatter.print_hash_diff(official_testcase_hashes, testcase_hashes)
+            if official_testcase_hashes != testcase_hashes:
```

Comment:
Maybe we can use `print_hash_diff_with_exit` here, just like `print_compile_string_with_exit`.